vispy/app/backends/_qt.py

```diff
@@ -199,15 +214,21 @@ def _set_config(c):
     glformat.setGreenBufferSize(c['green_size'])
     glformat.setBlueBufferSize(c['blue_size'])
     glformat.setAlphaBufferSize(c['alpha_size'])
-    glformat.setAccum(False)
-    glformat.setRgba(True)
-    glformat.setDoubleBuffer(True if c['double_buffer'] else False)
-    glformat.setDepth(True if c['depth_size'] else False)
+    if QT5_NEW_API:
+        # Qt5 >= 5.4.0 - below options automatically enabled if nonzero.
+        glformat.setSwapBehavior(glformat.DoubleBuffer if c['double_buffer']
```

Comment:
This is the "modern" sort of stuff I was talking about -- the new API introduced in #960

Reply:
I still a little confused. This PR was started by you @larsoner. I simply took what you had and modified a few things to get the tests to pass. The main thing being QOffscreenSurface use for rendering and getting a GL context without explicitly showing a Widget first.

This PR should be a replacement for #960 based on previous conversations we've had.